moveit_kinematics/test/test_kinematics_plugin.cpp

```diff
@@ -98,58 +94,41 @@ class SharedData
     ASSERT_TRUE(bool(robot_model_)) << "Failed to load robot model";
 
     // init ClassLoader
     kinematics_loader_ = std::make_unique<KinematicsLoader>("moveit_core", "kinematics::KinematicsBase");
     ASSERT_TRUE(bool(kinematics_loader_)) << "Failed to instantiate ClassLoader";
 
     // load parameters
-    ASSERT_TRUE(getParam(node_, "group", group_name_));
-    ASSERT_TRUE(getParam(node_, "tip_link", tip_link_));
-    ASSERT_TRUE(getParam(node_, "root_link", root_link_));
-    ASSERT_TRUE(getParam(node_, "joint_names", joints_));
-    getParam(node_, "seed", seed_);
-    ASSERT_TRUE(seed_.empty() || seed_.size() == joints_.size());
-    getParam(node_, "consistency_limits", consistency_limits_);
-    if (!getParam(node_, "ik_timeout", timeout_) || timeout_ < 0.0)
-      timeout_ = 1.0;
-    if (!getParam(node_, "tolerance", tolerance_) || tolerance_ < 0.0)
-      tolerance_ = DEFAULT_TOLERANCE;
-    ASSERT_TRUE(consistency_limits_.empty() || consistency_limits_.size() == joints_.size());
-    ASSERT_TRUE(getParam(node_, "num_fk_tests", num_fk_tests_));
-    ASSERT_TRUE(getParam(node_, "num_ik_cb_tests", num_ik_cb_tests_));
-    ASSERT_TRUE(getParam(node_, "num_ik_tests", num_ik_tests_));
-    ASSERT_TRUE(getParam(node_, "num_ik_multiple_tests", num_ik_multiple_tests_));
-    ASSERT_TRUE(getParam(node_, "num_nearest_ik_tests", num_nearest_ik_tests_));
+    node_->get_parameter_or("group", group_name_, UNDEFINED);
+    node_->get_parameter_or("tip_link", tip_link_, UNDEFINED);
+    node_->get_parameter_or("root_link", root_link_, UNDEFINED);
+    node_->get_parameter_or("joint_names", joints_, joints_);
+    node_->get_parameter_or("seed", seed_, { 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 });
+    ASSERT_TRUE(seed_.empty() || seed_.size() == joints_.size()) << "If set, 'seed' size must match 'joint_names' size";
+    node_->get_parameter_or("consistency_limits", consistency_limits_, consistency_limits_);
+    ASSERT_TRUE(consistency_limits_.empty() || consistency_limits_.size() == joints_.size())
+        << "If set, 'consistency_limits' size must match 'joint_names' size";
+    node_->get_parameter_or("ik_timeout", timeout_, 1.0);
+    ASSERT_TRUE(timeout_ > 0.0) << "'ik_timeout' must be more than 0.0 seconds";
+    node_->get_parameter_or("tolerance", tolerance_, tolerance_);
+    ASSERT_TRUE(tolerance_ > 0.0) << "'tolerance' must be greater than 0.0";
+    node_->get_parameter_or("num_fk_tests", num_fk_tests_, 0);
+    node_->get_parameter_or("num_ik_cb_tests", num_ik_cb_tests_, 0);
+    node_->get_parameter_or("num_ik_tests", num_ik_tests_, 0);
+    node_->get_parameter_or("num_ik_multiple_tests", num_ik_multiple_tests_, 0);
+    node_->get_parameter_or("num_nearest_ik_tests", num_nearest_ik_tests_, 0);
+    node_->get_parameter_or("ik_plugin_name", ik_plugin_name_, UNDEFINED);
+    node_->get_parameter_or("publish_trajectory", publish_trajectory_, false);
```

Comment:
I'm out of the loop on param retrieval and not sure how we want to standardize moveit2 on "this required param is not set in the namespace I'm expecting" - getParam is a nice pattern, but ends up being copied to a bunch of files, rather than being a single place "rosparam shortcuts package?"

Regardless, removing the getParam pattern + ASSERT_TRUE is not going to catch cases where a required param is not set. Do we have another test which will catch that? Or are these not truly "required params"? By required params I mean params which we cannot provide a "good" default value for.

Reply:
I am currently not aware if we have such tests that will catch this behavior other than getParam+ ASSERT_TRUE. I'll have to  look further for it. Maybe @tylerjw or @JafarAbdi knows? If we have such thing I would love to use it as well.